Jenkinsfile

```diff
@@ -500,6 +500,38 @@ lock(resource: "build-${params.STREAM}") {
             /var/tmp/fcos-releng/coreos-meta-translator/trans.py --workdir .
             """)
 
+            // Replace the 'next-release' field in 'release-notes.yaml' field
+            // with the real build id
+            botCreds = "github-coreosbot-token"
+            config_repo = "coreos/fedora-coreos-config"
+            release_notes_yaml = "src/config/release-notes.yaml"
+            utils.shwrap("""
+            /var/tmp/fcos-releng/coreos-release-note-generator/release-note-generator.py update \
+            --release-notes-file ${release_notes_yaml} \
+            --build-id ${buildID} \
+            --output-dir src/config
+            """)
+            shwrap("""
+            cd src/config
+            git commit -am "release-notes.yaml: replace 'next-release' with build id"
+            """)
+            withCredentials([usernamePassword(credentialsId: botCreds,
+                                                usernameVariable: 'GHUSER',
+                                                passwordVariable: 'GHTOKEN')]) {
+                // should gracefully handle race conditions here
+                sh("git push https://\${GHUSER}:\${GHTOKEN}@github.com/${config_repo} ${params.STREAM}")
+            }
+
+            // Push 'release-notes.yaml' to S3 bucket for website consumption
+            withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: 'aws-fcos-builds-bot']]) {
+                shwrap("""
+                cd src/config
+                aws s3 sync --acl public-read --cache-control 'max-age=60' \
+                    --exclude '*' --include 'release-notes.yaml' \
+                        ./ s3://fcos-builds
+                """)
+            }
+
```

Comment:
Something in my mind keeps telling me this is not correct / desired. Since `Jenkinsfile` primarily used _only_ used `cosa`, and not sure if I could use `botCreds` and `aws-fcos-builds-bot` here. The logic comes from `bump-lockfile.Jenkinsfile` and `sync-stream-metadata.Jenkinsfile`.

Could you sanity check please? @jlebon

Reply:
Still trying to understand how everything fits together (so the next statements might not fit with the intended design), but ISTM like we shouldn't do anything related to release notes as part of the main build pipeline. That should instead be in the release job (or a job triggered by the release job), and only on production streams.

Reply:
Agreed. Discussed at https://github.com/coreos/fedora-coreos-pipeline/pull/261#issuecomment-675104281.